generators/app/templates/src/module.h

```diff
@@ -5,11 +5,11 @@
 class <%= moduleClassName %> : public Napi::ObjectWrap<<%= moduleClassName %>>
 {
 public:
-    static void Initialize(Napi::Env&, Napi::Object&);
-
     <%= moduleClassName %>(const Napi::CallbackInfo&);
     Napi::Value Greet(const Napi::CallbackInfo&);
 
+    static Napi::Function GetClass(Napi::Env);
+
 private:
-    Napi::Reference<Napi::String> _greeterName;
-};
\ No newline at end of file
+    std::string _greeterName;
```

Comment:
What was the issue you ran into if this was a Napi::Reference? Although I don't feel very strongly that it should be a N-API string- I see your point that it's a good example of showing interop between C++ types and N-API types

Reply:
Calling ```Persistent``` function with a ```Napi::String``` to create the reference threw "An object was expected" exception.